packages/core-bridge/src/client.rs

```diff
@@ -711,4 +741,31 @@ mod config {
             }
         }
     }
+
+    fn partition_headers(
+        headers: Option<HashMap<String, MetadataValue>>,
+    ) -> (
+        Option<HashMap<String, String>>,
+        Option<HashMap<String, Vec<u8>>>,
+    ) {
+        let Some(headers) = headers else {
+            return (None, None);
+        };
+        let mut ascii_headers = HashMap::default();
```

Comment:
Possible optimization is to create the hashmap with the same capacity as the initial headers map. Only downside I could see would be if the headers were mostly binary.

Reply:
This is once per native client + on `clientUpdateHeader`. Neither of them should be called frequently enough to justify efforts on performance optimization IMO.

But if we'd indeed want to go there, I'd initiate _both_ hashmaps at the capacity of the initial headers map, up to a certain upper limit (something like 128 or 256). That should optimize performance in all cases while providing a reasonable upper limit on how much memory we may allocate uselessly.

But honestly, I wouldn't mind that one.